File: t3/viabnb.tier3.dataserver.rest/src/main/java/dk/viabnb/sep3/group6/dataserver/rest/t3/models/ResidenceReview.java
package dk.viabnb.sep3.group6.dataserver.rest.t3.models;

import java.time.LocalDate;

public class ResidenceReview
{
  private double rating;
  private String reviewText;
  private int guestViaId;
  private LocalDate createdDate;

  public ResidenceReview(double rating, String reviewText, int guestViaId, LocalDate createdDate) {
    this.rating = rating;
    this.reviewText = reviewText;
    this.guestViaId = guestViaId;
    this.createdDate = createdDate;
  }

  public double getRating()
  {
    return rating;
  }

  public void setRating(double rating)
  {
    this.rating = rating;
  }

  public String getReviewText()
  {
    return reviewText;
  }

  public void setReviewText(String reviewText)
  {
    this.reviewText = reviewText;
  }

  public int getGuestViaId()
  {
    return guestViaId;
  }

  public void setGuestViaId(int guestViaId)
  {
    this.guestViaId = guestViaId;
  }

  public LocalDate getCreatedDate() {
    return createdDate;
  }

  public void setCreatedDate(LocalDate createdDate) {
    this.createdDate = createdDate;
  }
}

File: t1/SEP3BlazorT1Client/SEP3BlazorT1Client/Models/ResidenceReview.cs
using System;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace SEP3BlazorT1Client.Models
{
    public class ResidenceReview
    {
        [Required]
        [JsonProperty("rating")]
        public double Rating { get; set; }
        [JsonProperty("reviewText")]
        public string ReviewText { get; set; }
        [JsonProperty("guestViaId")]
        [Required]
        public int GuestViaId { get; set; }

        [JsonProperty("createdDate")]
        public DateTime CreatedDate { get; set; }
    }
}